From 7d0c41b6236ec750229f3f5595c2b6fec69f2f8e Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 18:20:05 +0200
Subject: [PATCH] Name change


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index 24f390b..582b271 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -257,7 +257,7 @@ class Server{
 	 * @return string
 	 */
 	public function getName(){
-		return "PocketMine-MP";
+		return "PocketMine-Soft";
 	}
 
 	/**
-- 
2.1.4

