From d6ebb31ce96c11cb673868f16d5db6b8154beddf Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 18:20:05 +0200
Subject: [PATCH] Name change


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index 8ae5dde..f86979b 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -209,7 +209,7 @@ class Server{
 	 * @return string
 	 */
 	public function getName(){
-		return "PocketMine-MP";
+		return "PocketMine-Soft";
 	}
 
 	/**
-- 
1.9.1

