From 674db4f9df92aeb9c6f159764acad0ea4509ce98 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 18:20:05 +0200
Subject: [PATCH] Name change


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index 799775a..98e6409 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -223,7 +223,7 @@ class Server{
 	 * @return string
 	 */
 	public function getName(){
-		return "PocketMine-MP";
+		return "PocketMine-Soft";
 	}
 
 	/**
-- 
2.1.0

