From ff94a9156fe961df97988456d3f6aab1ff805696 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 18:20:05 +0200
Subject: [PATCH] Name change


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index 52af8e3..fe0bc6e 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -250,7 +250,7 @@ class Server{
 	 * @return string
 	 */
 	public function getName(){
-		return "PocketMine-MP";
+		return "PocketMine-Soft";
 	}
 
 	/**
-- 
2.1.4

