From 1127f13217368f275a561d3d4493225315034bd0 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 18:20:05 +0200
Subject: [PATCH] Name change


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index 42ae494..c0a1205 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -253,7 +253,7 @@ class Server{
 	 * @return string
 	 */
 	public function getName(){
-		return "PocketMine-MP";
+		return "PocketMine-Soft";
 	}
 
 	/**
-- 
2.1.4

