From 5f16669c0709122693594163fa438b9ea59afc70 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 18:20:05 +0200
Subject: [PATCH] Name change


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index 082256e..9fb75a0 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -206,7 +206,7 @@ class Server{
 	 * @return string
 	 */
 	public function getName(){
-		return "PocketMine-MP";
+		return "PocketMine-Soft";
 	}
 
 	/**
-- 
1.9.1

