From 0fe4658b9ad32388cdf6afd2605fd8c9f7454f21 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 18:20:05 +0200
Subject: [PATCH] Name change


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index f90a4bc..43e365f 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -202,7 +202,7 @@ class Server{
 	 * @return string
 	 */
 	public function getName(){
-		return "PocketMine-MP";
+		return "PocketMine-Soft";
 	}
 
 	/**
-- 
1.9.1

