From c2737abd0334d9ecd0f737b9d9aa3270a67a6f27 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 18:20:05 +0200
Subject: [PATCH] Name change


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index e246cdf..cabac66 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -224,7 +224,7 @@ class Server{
 	 * @return string
 	 */
 	public function getName(){
-		return "PocketMine-MP";
+		return "PocketMine-Soft";
 	}
 
 	/**
-- 
2.1.0

