From 067fb29c8c5e026b25c68193f42408fb548c6bfa Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 18:20:05 +0200
Subject: [PATCH] Name change


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index 049f4ac..a8bf1c8 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -222,7 +222,7 @@ class Server{
 	 * @return string
 	 */
 	public function getName(){
-		return "PocketMine-MP";
+		return "PocketMine-Soft";
 	}
 
 	/**
-- 
1.9.1

