From 6199f81cd0afc6b04bf27bd7a04579d49b761a5d Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 22 Aug 2014 19:38:07 +0200
Subject: [PATCH] Modified script names, disabled installer


diff --git a/src/pocketmine/PocketMine.php b/src/pocketmine/PocketMine.php
index d44e1ec..41fc5f9 100644
--- a/src/pocketmine/PocketMine.php
+++ b/src/pocketmine/PocketMine.php
@@ -69,7 +69,6 @@ namespace pocketmine {
 	use pocketmine\utils\Binary;
 	use pocketmine\utils\MainLogger;
 	use pocketmine\utils\Utils;
-	use pocketmine\wizard\Installer;
 	use raklib\RakLib;
 
 	const VERSION = "Alpha_1.4dev";
@@ -266,7 +265,7 @@ namespace pocketmine {
 	}
 
 	if(php_sapi_name() !== "cli"){
-		$logger->critical("You must run PocketMine-MP using the CLI.");
+		$logger->critical("You must run PocketMine-Soft using the CLI.");
 		++$errors;
 	}
 
@@ -340,12 +339,8 @@ namespace pocketmine {
 	@define("INT32_MASK", is_int(0xffffffff) ? 0xffffffff : -1);
 	@ini_set("opcache.mmap_base", bin2hex(Utils::getRandomBytes(8, false))); //Fix OPCache address errors
 
-	if(!file_exists(\pocketmine\DATA . "server.properties") and !isset($opts["no-wizard"])){
-		new Installer();
-	}
-
 	if(substr(__FILE__, 0, 7) !== "phar://"){
-		$logger->warning("Non-packaged PocketMine-MP installation detected, do not use on production.");
+		$logger->warning("Non-packaged PocketMine-Soft installation detected, do not use on production.");
 	}
 
 	ThreadManager::init();
diff --git a/start.cmd b/start.cmd
index a362b7c..d777993 100644
--- a/start.cmd
+++ b/start.cmd
@@ -7,8 +7,8 @@ if exist bin\php\php.exe (
 	set PHP_BINARY=php
 )
 
-if exist PocketMine-MP.phar (
-	set POCKETMINE_FILE=PocketMine-MP.phar
+if exist PocketMine-Soft.phar (
+	set POCKETMINE_FILE=PocketMine-Soft.phar
 ) else (
 	if exist src\pocketmine\PocketMine.php (
 		set POCKETMINE_FILE=src\pocketmine\PocketMine.php
diff --git a/start.sh b/start.sh
index 6a0c7b4..f6ae4d1 100755
--- a/start.sh
+++ b/start.sh
@@ -33,12 +33,12 @@ if [ "$PHP_BINARY" == "" ]; then
 fi
 
 if [ "$POCKETMINE_FILE" == "" ]; then
-	if [ -f ./PocketMine-MP.phar ]; then
-		POCKETMINE_FILE="./PocketMine-MP.phar"
+	if [ -f ./PocketMine-Soft.phar ]; then
+		POCKETMINE_FILE="./PocketMine-Soft.phar"
 	elif [ -f ./src/pocketmine/PocketMine.php ]; then
 		POCKETMINE_FILE="./src/pocketmine/PocketMine.php"
 	else
-		echo "Couldn't find a valid PocketMine-MP installation"
+		echo "Couldn't find a valid PocketMine-Soft installation"
 		exit 1
 	fi
 fi
-- 
1.9.1

