From a760e11f81d2604ab717aa53269a686bc2a19912 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Sat, 23 Aug 2014 00:59:36 +0200
Subject: [PATCH] Added filter-creative-items setting


diff --git a/src/pocketmine/Player.php b/src/pocketmine/Player.php
index ba73987..5588ce8 100644
--- a/src/pocketmine/Player.php
+++ b/src/pocketmine/Player.php
@@ -1107,6 +1107,11 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 	}
 
 	protected function getCreativeBlock(Item $item){
+
+		if(!$this->getServer()->getAdvancedProperty("main.filter-creative-items", true)){
+			return 0;
+		}
+
 		foreach(Block::$creative as $i => $d){
 			if($d[0] === $item->getID() and $d[1] === $item->getDamage()){
 				return $i;
diff --git a/src/pocketmine/resources/pocketmine-soft.yml b/src/pocketmine/resources/pocketmine-soft.yml
index 766c6bd..bb4257c 100644
--- a/src/pocketmine/resources/pocketmine-soft.yml
+++ b/src/pocketmine/resources/pocketmine-soft.yml
@@ -14,4 +14,8 @@ messages:
 
  #Message shown by default when a player doesn't have permissions to use a command
  #Can be overriden by the command
- command-permissions: "§cYou don't have permissions to use this command."
\ No newline at end of file
+ command-permissions: "§cYou don't have permissions to use this command."
+
+main:
+ #Block unknown items from creative-mode players
+ filter-creative-items: true
\ No newline at end of file
-- 
1.9.1

