From de914376efdab82821489d20c986aa19a8f29e3d Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Sat, 23 Aug 2014 01:06:28 +0200
Subject: [PATCH] Added shuffle-players setting


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index 51cb565..fe5067c 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -1761,6 +1761,10 @@ class Server{
 
 		$this->enablePlugins(PluginLoadOrder::POSTWORLD);
 
+		if($this->getAdvancedProperty("main.player-shuffle", 0) > 0){
+			$this->scheduler->scheduleDelayedRepeatingTask(new CallbackTask([$this, "shufflePlayers"]), $this->getAdvancedProperty("main.player-shuffle", 0), $this->getAdvancedProperty("main.player-shuffle", 0));
+		}
+
 		$this->start();
 	}
 
@@ -2353,4 +2357,19 @@ class Server{
 		Tile::registerTile(Sign::class);
 	}
 
+	public function shufflePlayers(){
+		if(count($this->players) <= 1){
+			return;
+		}
+
+		$keys = array_keys($this->players);
+		shuffle($keys);
+		$random = [];
+		foreach ($keys as $key) {
+			$random[$key] = $this->players[$key];
+		}
+
+		$this->players = $random;
+	}
+
 }
diff --git a/src/pocketmine/resources/pocketmine-soft.yml b/src/pocketmine/resources/pocketmine-soft.yml
index bb4257c..9302078 100644
--- a/src/pocketmine/resources/pocketmine-soft.yml
+++ b/src/pocketmine/resources/pocketmine-soft.yml
@@ -18,4 +18,8 @@ messages:
 
 main:
  #Block unknown items from creative-mode players
- filter-creative-items: true
\ No newline at end of file
+ filter-creative-items: true
+ 
+ #Shuffle the player list to prevent players from 'gaming' the server, getting high priority on the tick list.
+ #Value is in ticks, so 20 = shuffle each second
+ player-shuffle: 0
\ No newline at end of file
-- 
2.1.4

