From dce09698108b0582d86959f40f8d2cbeee1830e1 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Sat, 23 Aug 2014 01:30:53 +0200
Subject: [PATCH] Enabled auto-updater


diff --git a/src/pocketmine/Server.php b/src/pocketmine/Server.php
index 500bdf4..4330f04 100644
--- a/src/pocketmine/Server.php
+++ b/src/pocketmine/Server.php
@@ -1530,9 +1530,6 @@ class Server{
 		$this->logger->info("Loading pocketmine.yml...");
 		if(!file_exists($this->dataPath . "pocketmine.yml")){
 			$content = file_get_contents($this->filePath . "src/pocketmine/resources/pocketmine.yml");
-			if($version->isDev()){
-				$content = str_replace("preferred-channel: stable", "preferred-channel: beta", $content);
-			}
 			@file_put_contents($this->dataPath . "pocketmine.yml", $content);
 		}
 		$this->config = new Config($this->dataPath . "pocketmine.yml", Config::YAML, []);
diff --git a/src/pocketmine/resources/pocketmine.yml b/src/pocketmine/resources/pocketmine.yml
index 9edc9cb..0c9386d 100644
--- a/src/pocketmine/resources/pocketmine.yml
+++ b/src/pocketmine/resources/pocketmine.yml
@@ -81,14 +81,10 @@ auto-report:
  host: crash.pocketmine.net
 
 auto-updater:
- enabled: false
+ enabled: true
  on-update:
   warn-console: true
   warn-ops: true
- #Can be development, beta or stable.
- preferred-channel: stable
- #If using a development version, it will suggest changing the channel
- suggest-channels: true
  host: www.pocketmine.net
 
 aliases:
diff --git a/src/pocketmine/updater/AutoUpdater.php b/src/pocketmine/updater/AutoUpdater.php
index 1b3e9f9..0928c35 100644
--- a/src/pocketmine/updater/AutoUpdater.php
+++ b/src/pocketmine/updater/AutoUpdater.php
@@ -45,13 +45,6 @@ class AutoUpdater{
 				if($this->server->getProperty("auto-updater.on-update.warn-console", true)){
 					$this->showConsoleUpdate();
 				}
-			}elseif($this->server->getProperty("auto-updater.preferred-channel", true)){
-				$version = new VersionString();
-				if(!$version->isDev() and $this->getChannel() !== "stable"){
-					$this->showChannelSuggestionStable();
-				}elseif($version->isDev() and $this->getChannel() === "stable"){
-					$this->showChannelSuggestionBeta();
-				}
 			}
 		}
 	}
@@ -85,34 +78,34 @@ class AutoUpdater{
 	public function showConsoleUpdate(){
 		$logger = $this->server->getLogger();
 		$newVersion = new VersionString($this->updateInfo["version"]);
-		$logger->warning("----- PocketMine-MP Auto Updater -----");
-		$logger->warning("Your version of PocketMine-MP is out of date. Version " . $newVersion->get(false) . " (build #" . $newVersion->getBuild() . ") was released on " . date("D M j h:i:s Y", $this->updateInfo["date"]));
+		$logger->warning("----- PocketMine-Soft Auto Updater -----");
+		$logger->warning("Your version of PocketMine-SoftP is out of date. Version " . $newVersion->get(false) . " (build #" . $newVersion->getBuild() . ") was released on " . date("D M j h:i:s Y", $this->updateInfo["date"]));
 		if($this->updateInfo["details_url"] !== null){
 			$logger->warning("Details: " . $this->updateInfo["details_url"]);
 		}
 		$logger->warning("Download: " . $this->updateInfo["download_url"]);
-		$logger->warning("----- -------------------------- -----");
+		$logger->warning("----- ---------------------------- -----");
 	}
 
 	public function showPlayerUpdate(Player $player){
-		$player->sendMessage(TextFormat::DARK_PURPLE . "The version of PocketMine-MP that this server is running is out of date. Please consider updating to the latest version.");
+		$player->sendMessage(TextFormat::DARK_PURPLE . "The version of PocketMine-Soft that this server is running is out of date. Please consider updating to the latest version.");
 		$player->sendMessage(TextFormat::DARK_PURPLE . "Check the console for more details.");
 	}
 
 	protected function showChannelSuggestionStable(){
 		$logger = $this->server->getLogger();
-		$logger->info("----- PocketMine-MP Auto Updater -----");
+		$logger->info("----- PocketMine-Soft Auto Updater -----");
 		$logger->info("It appears you're running a Stable build, when you've specified that you prefer to run " . ucfirst($this->getChannel()) . " builds.");
 		$logger->info("If you would like to be kept informed about new Stable builds only, it is recommended that you change 'preferred-channel' in your pocketmine.yml to 'stable'.");
-		$logger->info("----- -------------------------- -----");
+		$logger->info("----- ---------------------------- -----");
 	}
 
 	protected function showChannelSuggestionBeta(){
 		$logger = $this->server->getLogger();
-		$logger->info("----- PocketMine-MP Auto Updater -----");
+		$logger->info("----- PocketMine-Soft Auto Updater -----");
 		$logger->info("It appears you're running a Beta build, when you've specified that you prefer to run Stable builds.");
 		$logger->info("If you would like to be kept informed about new Beta or Development builds, it is recommended that you change 'preferred-channel' in your pocketmine.yml to 'beta' or 'development'.");
-		$logger->info("----- -------------------------- -----");
+		$logger->info("----- ---------------------------- -----");
 	}
 
 	public function getUpdateInfo(){
@@ -139,11 +132,6 @@ class AutoUpdater{
 	}
 
 	public function getChannel(){
-		$channel = strtolower($this->server->getProperty("auto-updater.preferred-channel", "stable"));
-		if($channel !== "stable" and $channel !== "beta" and $channel !== "development"){
-			$channel = "stable";
-		}
-
-		return $channel;
+		return "soft";
 	}
 }
\ No newline at end of file
-- 
2.1.0

