From f011f6459f8fd8368d6b5647d29a1a87b9921a59 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Sun, 24 Aug 2014 17:26:54 +0200
Subject: [PATCH] Added RakLib port checking option


diff --git a/src/pocketmine/network/RakLibInterface.php b/src/pocketmine/network/RakLibInterface.php
index 000fb91..ea787aa 100644
--- a/src/pocketmine/network/RakLibInterface.php
+++ b/src/pocketmine/network/RakLibInterface.php
@@ -66,6 +66,11 @@ class RakLibInterface implements ServerInstance, AdvancedSourceInterface{
 		$this->rakLib = new RakLibServer($this->server->getLogger(), $this->server->getLoader(), $this->server->getPort(), $this->server->getIp() === "" ? "0.0.0.0" : $this->server->getIp());
 		$this->interface = new ServerHandler($this->rakLib, $this);
 
+		if(!$this->server->getAdvancedProperty("main.session-port-checking", true)){
+			$this->setPortCheck(false);
+		}
+
+
 		for($i = 0; $i < 256; ++$i){
 			$this->channelCounts[$i] = 0;
 		}
diff --git a/src/pocketmine/resources/pocketmine-soft.yml b/src/pocketmine/resources/pocketmine-soft.yml
index a7be578..f6951e8 100644
--- a/src/pocketmine/resources/pocketmine-soft.yml
+++ b/src/pocketmine/resources/pocketmine-soft.yml
@@ -23,3 +23,7 @@ main:
  #Shuffle the player list to prevent players from 'gaming' the server, getting high priority on the tick list.
  #Value is in ticks, so 20 = shuffle each second
  player-shuffle: 0
+
+ #Enables RakLib client/server port checking
+ #Proxied servers might have to disable this option
+ session-port-checking: true
-- 
2.1.4

