From 0ef4e5b6396a5b25631f9afb00398dd083d6bcc0 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Tue, 26 Aug 2014 17:44:20 +0200
Subject: [PATCH] Disable chunk ticking by default


diff --git a/src/pocketmine/level/Level.php b/src/pocketmine/level/Level.php
index 6cffa8d..2a7ed72 100644
--- a/src/pocketmine/level/Level.php
+++ b/src/pocketmine/level/Level.php
@@ -295,7 +295,7 @@ class Level implements ChunkManager, Metadatable{
 		$this->time = (int) $this->provider->getTime();
 
 		$this->chunkTickRadius = min($this->server->getViewDistance(), max(1, (int) $this->server->getProperty("chunk-ticking.tick-radius", 4)));
-		$this->chunksPerTick = (int) $this->server->getProperty("chunk-ticking.per-tick", 260);
+		$this->chunksPerTick = (int) $this->server->getProperty("chunk-ticking.per-tick", 0);
 		$this->chunkTickList = [];
 		$this->clearChunksOnTick = (bool) $this->server->getProperty("chunk-ticking.clear-tick-list", false);
 
diff --git a/src/pocketmine/resources/pocketmine.yml b/src/pocketmine/resources/pocketmine.yml
index 1e1ae86..a1874a3 100644
--- a/src/pocketmine/resources/pocketmine.yml
+++ b/src/pocketmine/resources/pocketmine.yml
@@ -41,7 +41,7 @@ chunk-sending:
 
 chunk-ticking:
  #Max amount of chunks processed each tick
- per-tick: 260
+ per-tick: 0
  #Radius of chunks around a player to tick
  tick-radius: 4
  #NOTE: This is currently not implemented
-- 
2.1.0

