From 1bd7c2107ca70933b0f359d8448794b818a1a1d0 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Tue, 26 Aug 2014 17:44:20 +0200
Subject: [PATCH] Disable chunk ticking by default


diff --git a/src/pocketmine/level/Level.php b/src/pocketmine/level/Level.php
index 88c1604..07697fd 100644
--- a/src/pocketmine/level/Level.php
+++ b/src/pocketmine/level/Level.php
@@ -261,7 +261,7 @@ class Level implements ChunkManager, Metadatable{
 		$this->time = (int) $this->provider->getTime();
 
 		$this->chunkTickRadius = min($this->server->getViewDistance(), max(1, (int) $this->server->getProperty("chunk-ticking.tick-radius", 3)));
-		$this->chunksPerTick = (int) $this->server->getProperty("chunk-ticking.per-tick", 80);
+		$this->chunksPerTick = (int) $this->server->getProperty("chunk-ticking.per-tick", 0);
 		$this->chunkTickList = [];
 		$this->clearChunksOnTick = (bool) $this->server->getProperty("chunk-ticking.clear-tick-list", false);
 
diff --git a/src/pocketmine/resources/pocketmine.yml b/src/pocketmine/resources/pocketmine.yml
index 10c5fb0..f0cd2ae 100644
--- a/src/pocketmine/resources/pocketmine.yml
+++ b/src/pocketmine/resources/pocketmine.yml
@@ -40,7 +40,7 @@ chunk-sending:
 
 chunk-ticking:
  #Max amount of chunks processed each tick
- per-tick: 80
+ per-tick: 0
  #Radius of chunks around a player to tick
  tick-radius: 2
  light-updates: false
-- 
1.9.1

