From c66553a709d076d323af610caf20ddae5e43149c Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Mon, 1 Sep 2014 18:32:00 +0200
Subject: [PATCH] Fixed RakLib interface start


diff --git a/src/pocketmine/network/RakLibInterface.php b/src/pocketmine/network/RakLibInterface.php
index a490765..ff56ba9 100644
--- a/src/pocketmine/network/RakLibInterface.php
+++ b/src/pocketmine/network/RakLibInterface.php
@@ -120,8 +120,7 @@ class RakLibInterface implements ServerInstance, SourceInterface{
 		$server = new RakLibServer($this->internalThreaded, $this->externalThreaded, $this->server->getLogger(), $this->server->getLoader(), $this->server->getPort(), $this->server->getIp() === "" ? "0.0.0.0" : $this->server->getIp());
 		$this->interface = new ServerHandler($server, $this);
 		$this->setName($this->server->getMotd());
-
-		if(!$server->getAdvancedProperty("main.session-port-checking", true)){
+		if(!$this->server->getAdvancedProperty("main.session-port-checking", true)){
 			$this->setPortCheck(false);
 		}
 
@@ -348,4 +347,4 @@ class RakLibInterface implements ServerInstance, SourceInterface{
 
 		return $data;
 	}
-}
\ No newline at end of file
+}
-- 
2.1.0

