From 9d59b50415836da402dbf3931c34e1df59c3c6ee Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Mon, 1 Sep 2014 18:32:00 +0200
Subject: [PATCH] Fixed RakLib interface start


diff --git a/src/pocketmine/network/RakLibInterface.php b/src/pocketmine/network/RakLibInterface.php
index cd3735f..07f37a5 100644
--- a/src/pocketmine/network/RakLibInterface.php
+++ b/src/pocketmine/network/RakLibInterface.php
@@ -117,8 +117,7 @@ class RakLibInterface implements ServerInstance, SourceInterface{
 		$server = new RakLibServer($this->internalThreaded, $this->externalThreaded, $this->server->getLogger(), $this->server->getLoader(), $this->server->getPort(), $this->server->getIp() === "" ? "0.0.0.0" : $this->server->getIp());
 		$this->interface = new ServerHandler($server, $this);
 		$this->setName($this->server->getMotd());
-
-		if(!$server->getAdvancedProperty("main.session-port-checking", true)){
+		if(!$this->server->getAdvancedProperty("main.session-port-checking", true)){
 			$this->setPortCheck(false);
 		}
 
@@ -345,4 +344,4 @@ class RakLibInterface implements ServerInstance, SourceInterface{
 
 		return $data;
 	}
-}
\ No newline at end of file
+}
-- 
1.9.1

