From a1bd90efa8dac147364303ea696a61a1a69ec8a7 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Mon, 13 Oct 2014 23:23:00 +0200
Subject: [PATCH] Added check-block-collision property


diff --git a/src/pocketmine/entity/Entity.php b/src/pocketmine/entity/Entity.php
index a91476b..76f74f2 100644
--- a/src/pocketmine/entity/Entity.php
+++ b/src/pocketmine/entity/Entity.php
@@ -101,6 +101,9 @@ abstract class Entity extends Location implements Metadatable{
 	private static $knownEntities = [];
 	private static $shortNames = [];
 
+	protected $checkBlockCollisionCount = 0;
+	protected $checkBlockCollisionTicks = 1;
+
 	/**
 	 * @var Player[]
 	 */
@@ -270,6 +273,8 @@ abstract class Entity extends Location implements Metadatable{
 		$this->lastUpdate = $this->server->getTick();
 		$this->server->getPluginManager()->callEvent(new EntitySpawnEvent($this));
 
+		$this->checkBlockCollisionTicks = (int) $this->server->getAdvancedProperty("main.check-block-collision", 1);
+
 		$this->scheduleUpdate();
 
 	}
@@ -791,7 +796,10 @@ abstract class Entity extends Location implements Metadatable{
 
 		$hasUpdate = false;
 
-		$this->checkBlockCollision();
+		if($this->checkBlockCollisionTicks > 0 and ++$this->checkBlockCollisionCount >= $this->checkBlockCollisionTicks){
+			$this->checkBlockCollisionCount = 0;
+			$this->checkBlockCollision();
+		}
 
 		if($this->y <= -16 and $this->isAlive()){
 			$ev = new EntityDamageEvent($this, EntityDamageEvent::CAUSE_VOID, 10);
diff --git a/src/pocketmine/resources/pocketmine-soft.yml b/src/pocketmine/resources/pocketmine-soft.yml
index 8415eaa..c23fd27 100644
--- a/src/pocketmine/resources/pocketmine-soft.yml
+++ b/src/pocketmine/resources/pocketmine-soft.yml
@@ -27,6 +27,10 @@ main:
  #If disabled, no checks for block collision will be done on players.
  check-movement: true
 
+ #Do entity block collision checks. This includes damage by blocks, falling on water, climbing blocks
+ #Value is in ticks, so 1 = check each tick. If 0, disable collision checks
+ check-block-collision: 1
+
  #Enables RakLib client/server port checking
  #Proxied servers might have to disable this option
  session-port-checking: true
-- 
2.1.4

