From 1623d9606f56170caeb5bfedc0680d3b7b63410b Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Sun, 5 Oct 2014 16:41:04 +0200
Subject: [PATCH] Added check-movement property


diff --git a/src/pocketmine/Player.php b/src/pocketmine/Player.php
index a7c47ff..d31111c 100644
--- a/src/pocketmine/Player.php
+++ b/src/pocketmine/Player.php
@@ -184,6 +184,8 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 
 	protected $inAirTicks = 0;
 
+	private $checkMovement;
+
 
 	private $needACK = [];
 
@@ -426,6 +428,7 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 		$this->setLevel($this->server->getDefaultLevel(), true);
 		$this->viewDistance = $this->server->getViewDistance();
 		$this->newPosition = new Vector3(0, 0, 0);
+		$this->checkMovement = (bool) $this->server->getAdvancedProperty("main.check-movement", true);
 	}
 
 	/**
@@ -1094,7 +1097,7 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 
 			$diff = $diffX ** 2 + $diffY ** 2 + $diffZ ** 2;
 
-			if(!$revert and !$this->isSleeping() and $this->isSurvival()){
+			if(!$revert and !$this->isSleeping() and ($this->isSurvival() and $this->checkMovement)){
 				if($diff > 0.0625){
 					$revert = true;
 					//$this->server->getLogger()->warning($this->getName()." moved wrongly!");
diff --git a/src/pocketmine/resources/pocketmine-soft.yml b/src/pocketmine/resources/pocketmine-soft.yml
index 7efb1f4..8a1d394 100644
--- a/src/pocketmine/resources/pocketmine-soft.yml
+++ b/src/pocketmine/resources/pocketmine-soft.yml
@@ -24,6 +24,10 @@ main:
  #Value is in ticks, so 20 = shuffle each second
  player-shuffle: 0
 
+ #Checks movement of players in survival mode and reverts them if not valid
+ #Creative mode is not affected
+ check-movement: true
+
  #Enables RakLib client/server port checking
  #Proxied servers might have to disable this option
  session-port-checking: true
-- 
1.9.1

