From 55f9be1cc5928e537485c384ad11e162f7880365 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Sun, 5 Oct 2014 16:41:04 +0200
Subject: [PATCH] Added check-movement property


diff --git a/src/pocketmine/Player.php b/src/pocketmine/Player.php
index c311ada..815312e 100644
--- a/src/pocketmine/Player.php
+++ b/src/pocketmine/Player.php
@@ -192,6 +192,7 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 
 	protected $inAirTicks = 0;
 
+	private $checkMovement;
 
 	private $needACK = [];
 
@@ -434,6 +435,7 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 		$this->setLevel($this->server->getDefaultLevel(), true);
 		$this->viewDistance = $this->server->getViewDistance();
 		$this->newPosition = new Vector3(0, 0, 0);
+		$this->checkMovement = (bool) $this->server->getAdvancedProperty("main.check-movement", true);
 		$this->boundingBox = new AxisAlignedBB(0, 0, 0, 0, 0, 0);
 	}
 
@@ -1115,10 +1117,10 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 			$diff = $diffX ** 2 + $diffY ** 2 + $diffZ ** 2;
 
 			if(!$revert and !$this->isSleeping() and $this->isSurvival()){
-				if($diff > 0.0625){
+				if($diff > 0.0625 and $this->checkMovement){
 					$revert = true;
 					//$this->server->getLogger()->warning($this->getName()." moved wrongly!");
-				}elseif($diff > 0){
+				}elseif($diff > 0 or !$this->checkMovement){
 					$this->x = $this->newPosition->x;
 					$this->y = $this->newPosition->y;
 					$this->z = $this->newPosition->z;
diff --git a/src/pocketmine/resources/pocketmine-soft.yml b/src/pocketmine/resources/pocketmine-soft.yml
index 7efb1f4..8a1d394 100644
--- a/src/pocketmine/resources/pocketmine-soft.yml
+++ b/src/pocketmine/resources/pocketmine-soft.yml
@@ -24,6 +24,10 @@ main:
  #Value is in ticks, so 20 = shuffle each second
  player-shuffle: 0
 
+ #Checks movement of players in survival mode and reverts them if not valid
+ #Creative mode is not affected
+ check-movement: true
+
  #Enables RakLib client/server port checking
  #Proxied servers might have to disable this option
  session-port-checking: true
-- 
2.1.0

