From 8212b28d49264eb5fd54d423d14f8dec6aa11968 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Mon, 13 Oct 2014 23:23:00 +0200
Subject: [PATCH] Added check-block-collision property


diff --git a/src/pocketmine/entity/Entity.php b/src/pocketmine/entity/Entity.php
index 645ec0c..51994de 100644
--- a/src/pocketmine/entity/Entity.php
+++ b/src/pocketmine/entity/Entity.php
@@ -70,6 +70,9 @@ abstract class Entity extends Location implements Metadatable{
 	/** @var Entity[] */
 	private static $knownEntities = [];
 
+	protected $checkBlockCollisionCount = 0;
+	protected $checkBlockCollisionTicks = 1;
+
 	/**
 	 * @var Player[]
 	 */
@@ -219,6 +222,8 @@ abstract class Entity extends Location implements Metadatable{
 		$this->lastUpdate = $this->server->getTick();
 		$this->server->getPluginManager()->callEvent(EntitySpawnEvent::createEvent($this));
 
+		$this->checkBlockCollisionTicks = (int) $this->server->getAdvancedProperty("main.check-block-collision", 1);
+
 		$this->scheduleUpdate();
 
 	}
@@ -519,7 +524,10 @@ abstract class Entity extends Location implements Metadatable{
 
 		$hasUpdate = false;
 
-		$this->checkBlockCollision();
+		if($this->checkBlockCollisionTicks > 0 and ++$this->checkBlockCollisionCount >= $this->checkBlockCollisionTicks){
+			$this->checkBlockCollisionCount = 0;
+			$this->checkBlockCollision();
+		}
 
 		if($this->y < 0 and $this->dead !== true){
 			$ev = EntityDamageEvent::createEvent($this, EntityDamageEvent::CAUSE_VOID, 10);
diff --git a/src/pocketmine/resources/pocketmine-soft.yml b/src/pocketmine/resources/pocketmine-soft.yml
index 8a1d394..136207b 100644
--- a/src/pocketmine/resources/pocketmine-soft.yml
+++ b/src/pocketmine/resources/pocketmine-soft.yml
@@ -28,6 +28,10 @@ main:
  #Creative mode is not affected
  check-movement: true
 
+ #Do entity block collision checks. This includes damage by blocks, falling on water, climbing blocks
+ #Value is in ticks, so 1 = check each tick. If 0, disable collision checks
+ check-block-collision: 1
+
  #Enables RakLib client/server port checking
  #Proxied servers might have to disable this option
  session-port-checking: true
-- 
1.9.1

