From f81592c3e833cfca438208acde13fae981fe1487 Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Fri, 16 Jan 2015 01:40:11 +0100
Subject: [PATCH] Added player view area checking and main.interaction-angle


diff --git a/src/pocketmine/Player.php b/src/pocketmine/Player.php
index cbd7029..f4a8757 100644
--- a/src/pocketmine/Player.php
+++ b/src/pocketmine/Player.php
@@ -1662,12 +1662,28 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 				$packet->eid = $this->id;
 
 				if($packet->face >= 0 and $packet->face <= 5){ //Use Block, place
+
+					$dir = $this->add(0, $this->getEyeHeight(), 0)->subtract($blockVector->add(0.5, 0.5, 0.5));
+
+					$len = $dir->length();
+					if($len === 0){
+						$len = 1;
+					}
+
+					$pitch = rad2deg(asin($dir->y / $len));
+					$yaw = rad2deg(atan2($dir->z, $dir->x)) + 90;
+
+					$viewAngle = (float) $this->getServer()->getAdvancedProperty("main.interaction-angle", 360);
+
+					$isYaw = ((180 - abs(abs($this->yaw - $yaw) - 180)) < $viewAngle);
+					$isPitch = ((180 - abs(abs($this->pitch - $pitch) - 180)) < $viewAngle);
+
 					if($this->inAction === true){
 						$this->inAction = false;
 						$this->sendMetadata($this->getViewers());
 					}
 
-					if($blockVector->distance($this) > 10){
+					if(!$isYaw or !$isPitch or $blockVector->distance($this) > 10){
 
 					}elseif($this->isCreative()){
 						$item = $this->inventory->getItemInHand();
@@ -1897,6 +1913,7 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 
 				$this->craftingType = 0;
 
+				/** @var Entity $cancelled */
 				$target = $this->level->getEntity($packet->target);
 
 				$cancelled = false;
@@ -1916,6 +1933,26 @@ class Player extends Human implements CommandSender, InventoryHolder, IPlayer{
 						return;
 					}
 
+					$dir = $this->add(0, $this->getEyeHeight(), 0)->subtract($target->add(0, $target->height / 2, 0));
+
+					$len = $dir->length();
+					if($len === 0){
+						$len = 1;
+					}
+
+					$pitch = rad2deg(asin($dir->y / $len));
+					$yaw = rad2deg(atan2($dir->z, $dir->x)) + 90;
+
+					$viewAngle = (float) $this->getServer()->getAdvancedProperty("main.interaction-angle");
+
+					$isYaw = ((180 - abs(abs($this->yaw - $yaw) - 180)) < $viewAngle);
+					$isPitch = ((180 - abs(abs($this->pitch - $pitch) - 180)) < $viewAngle);
+
+					if(!$isYaw or !$isPitch){
+						$cancelled = true;
+					}
+
+
 					$item = $this->inventory->getItemInHand();
 					$damageTable = [
 						Item::WOODEN_SWORD => 4,
diff --git a/src/pocketmine/resources/pocketmine-soft.yml b/src/pocketmine/resources/pocketmine-soft.yml
index d552658..ee6a120 100644
--- a/src/pocketmine/resources/pocketmine-soft.yml
+++ b/src/pocketmine/resources/pocketmine-soft.yml
@@ -28,6 +28,10 @@ main:
  #Creative mode is not affected
  check-movement: true
 
+ #Checks attacks and block placement to be in this view angle.
+ #360 is all, 0 none. A good value for cheat protection is 90
+ interaction-angle: 360
+
  #Do entity block collision checks. This includes damage by blocks, falling on water, climbing blocks
  #Value is in ticks, so 1 = check each tick. If 0, disable collision checks
  check-block-collision: 1
-- 
2.1.0

