From 3e4ea6f03d411fd4dff17ec80234a3e2ef2e3b8e Mon Sep 17 00:00:00 2001
From: Shoghi Cervantes <shoghicp@gmail.com>
Date: Mon, 13 Oct 2014 23:23:00 +0200
Subject: [PATCH] Added check-block-collision property


diff --git a/src/pocketmine/entity/Entity.php b/src/pocketmine/entity/Entity.php
index 3d63b12..edc3f2e 100644
--- a/src/pocketmine/entity/Entity.php
+++ b/src/pocketmine/entity/Entity.php
@@ -66,6 +66,9 @@ abstract class Entity extends Location implements Metadatable{
 
 	public static $entityCount = 1;
 
+	protected $checkBlockCollisionCount = 0;
+	protected $checkBlockCollisionTicks = 1;
+
 	/**
 	 * @var Player[]
 	 */
@@ -215,6 +218,8 @@ abstract class Entity extends Location implements Metadatable{
 		$this->lastUpdate = $this->server->getTick();
 		$this->server->getPluginManager()->callEvent(EntitySpawnEvent::createEvent($this));
 
+		$this->checkBlockCollisionTicks = (int) $this->server->getAdvancedProperty("main.check-block-collision", 1);
+
 		$this->scheduleUpdate();
 
 	}
@@ -480,7 +485,10 @@ abstract class Entity extends Location implements Metadatable{
 
 		$hasUpdate = false;
 
-		$this->checkBlockCollision();
+		if($this->checkBlockCollisionTicks > 0 and ++$this->checkBlockCollisionCount >= $this->checkBlockCollisionTicks){
+			$this->checkBlockCollisionCount = 0;
+			$this->checkBlockCollision();
+		}
 
 		if($this->y < 0 and $this->dead !== true){
 			$this->server->getPluginManager()->callEvent($ev = EntityDamageEvent::createEvent($this, EntityDamageEvent::CAUSE_VOID, 10));
diff --git a/src/pocketmine/resources/pocketmine-soft.yml b/src/pocketmine/resources/pocketmine-soft.yml
index 8a1d394..136207b 100644
--- a/src/pocketmine/resources/pocketmine-soft.yml
+++ b/src/pocketmine/resources/pocketmine-soft.yml
@@ -28,6 +28,10 @@ main:
  #Creative mode is not affected
  check-movement: true
 
+ #Do entity block collision checks. This includes damage by blocks, falling on water, climbing blocks
+ #Value is in ticks, so 1 = check each tick. If 0, disable collision checks
+ check-block-collision: 1
+
  #Enables RakLib client/server port checking
  #Proxied servers might have to disable this option
  session-port-checking: true
-- 
1.9.1

